design/ozone.md

```diff
@@ -0,0 +1,258 @@
+# Operator for Apache Hadoop Ozone
+
+## What is Ozone?
+
+Ozone is a new generic storage system. It is highly scalable and can be used via multiple interface:
+
+    It works as an S3 compatible file system
+    It can be used as a Hadoop compatible file system (from spark, hive, hbase, etc.)
+    It has an internal CSI server and can be used as raw kubernetes storage.
+
+You can find more information here: http://hadoop.apache.org/ozone/
+
+## Why?
+
+The two main advantages of using Ozone in kubernetes with Rook:
+
+ * It can provide Hadoop file system interface which is not provided by any other storage. Existing Hadoop compatible big data projects can use the storage via the Hadoop FS, but in the mean time the same data can be access from s3 interface or as a k8s volume.
+ * The design is based on the HDFS design. It can be scaled easily to thousand of nodes (but in the mean time the legendary small file problem of HDFS is also solved.)
+
+
+## Components of Ozone
+
+Ozone has the following components in a typical cluster:
+
+ * Two kind of metadata servers:
+    * **Storage Container Manager** (SCM): for block space management (create huge binary blocks which are replicated between the datanodes)
+    * **Ozone Manager** (OM): for key space management (create buckets, keys on top the blocks which are replicated by the Storage Container Manager)
+ * **Datanodes**: Storage daemons
+ * **S3 gateway** (stateless rest services, with AWS S3 compatible interface)
+ * **Recon** server (generic UI, anomaly detection, historical data)
+ * DB backend for recon
+
+The CSI interface requires two additional daemons:
+
+  * CSI Controller service
+  * CSI Node service
+  
+Ozone uses `StatefulSet` for the master servers (Storage Container Manager, Ozone Manager, S3 gateway, Recon Server) and slave servers (datanode)
+
+![Ozone components](https://i.imgur.com/e7Tmij4.png)
```

Comment:
I think we should put such images inside the Rook repo, as, e.g., the Rook Ceph diagrams.

Reply:
Sure. I added it to the PR.